odoo/tools/config.py

```diff
@@ -575,6 +577,32 @@ def die(cond, msg):
         ]
         return opt
 
+    def _validate_pg_config(self):
+        db_name = self.options["db_name"] or "postgres"
+        db = odoo.sql_db.db_connect(db_name)
+        with closing(db.cursor()) as cr:
+            if (pg_version := cr._cnx.server_version) < 15_0000:
```

Comment:
As we don't need specific features of 15.0, there is no need to enforce this version.
13.0 is still supported until November 2025, so why would we want to force people using that version to upgrade it before the end of life. Also, as you can see, we still have 14.0 versions on some runbot.
So IMHO, it's preferable to use 13.0 as minimal version.

Reply:
And so? We will still need to support Odoo 18 next year.
By forcing PG15, the EOL dates of Odoo and PG match nicely.

Reply:
I'm sorry but `Matching nicely` is not a requirement. It may bother a lot of people unnecessarily.

Reply:
> And so? 

- Noble is not yet the default version on dev machines, it is still jammy and psql 14. Noble is not considered stable yet regarding interface
- runbot is running on 3.14
- Minimal python version of odoo remains 3.10 (jammy) and so the corresponding psql is 14

We can advice to use psql 16 but for development there is no reason to forbid using 14.0. This would force to upgrade developer machine in a rush. All the machine of runbot must be upgraded first too witch will take some time.

Reply:
My 2cents: minimal PG version should be something with at least one year support at the release time of the new Odoo version _and_ at least higher than what's used in CI (saas and sh taken into account ofc). Otherwise we get the current situation where some queries may fail when the "documented" minimal version is used (i.e. during upgrades) while all CIs were green.

Thus PG14 looks like the best option to me.

Reply:
So for python package, the rule is "use the version of the Debian stable package", but for PostgreSQL this rule is not acceptable?
And using `pip` on dev machine is not an excuse. You can install multiple PG versions on a machine via the [PostgreSQL APT Repository](https://www.postgresql.org/download/linux/ubuntu/).

> and at least equal or higher than what's used in CI 

Sorry, but no. Our requirements should not be imposed by the CI infrastructure. The dependency should be in the other way around: If we need some specific minimal package version, that's the CI that needs to be adapted, not the version.

> Thus PG14 looks like the best option to me.

PG14 will be OEL on Nov 2026, while Odoo 18 will be EOL on Q4 2027.

Reply:
> So for python package, the rule is "use the version of the Debian stable package", 

This is not True. The rule is, use the version of the package installed by default on the os you are using.
What you are referring to is how we pin requirements, and even there we have conditional version depending in python version. 

The rule would be, for python3.10 use postgres14 and for python3.12 use postgres 16
But it is true that regarding testing the psql version used depends on the server version and not branch version like other packages. But we are still in the supposed to be supported ranges.

Reply:
> Sorry, but no. Our requirements should not be imposed by the CI infrastructure. The dependency should be in the other way around: If we need some specific minimal package version, that's the CI that needs to be adapted, not the version.

I agree. My 2cents were about the _practical_ reason we face right now: what was the default in those systems when the OS was released (14, if I read it correctly). If we want to be strict in terms of the compatibility of EOL dates then your suggestion seems right.